### PR TITLE
Improve Sentry

### PR DESCRIPTION
**Pull request by oscar-h64**
This PR implements request types 2 and 4 to sentry, and implements some (very ugly) renaming from new mode attributes to old mode attributes (based off the real rename function from slack)

Draft because the attributes look right, but I realised while writing this I haven't actually tested it works with Tabula, I'll do that tomorrow

### CODE REVIEW

**Review by lol768 — APPROVED** [review]
Thanks Oscar, this LGTM! Let us know how the Tabula testing goes

As a future development, it might be possible to stub out the user search endpoint (to return nothing) just to quash the flexipicker errors, too.

**Comment by oscar-h64** [comment]
Lucky I tested it, the credentials are supposed to be in the POST body not the parameters (obviously). That should be good now, I can make API requests to Tabula with it

File: app/controllers/IndexController.scala [hunk 1]
```diff
@@ -83,7 +85,7 @@ class IndexController extends BaseController {
     val reqId = (request.body \\ "Envelope" \\ "Body" \\ "Request" \\ "@RequestID" headOption).map(_.text)
     val member = (fakeMemberService.getStaff++fakeMemberService.getStudents).filter(m => m.universityId == name.get).head
 
-    val attributes: Seq[SAMLAttribute] = AttributeConverter.toAttributes(fakeMemberService.getResponseFor(member)).toList map {
+    val attributes: Seq[SAMLAttribute] = AttributeConverter.toAttributes(fakeMemberService.getResponseFor(member), false).toList map {
```
Comment (matmannion) [inline app/controllers/IndexController.scala]: Can we name this boolean parameter please?

File: app/controllers/IndexController.scala [hunk 2]
```diff
@@ -121,18 +123,34 @@ class IndexController extends BaseController {
...
-  def respondToSentry() = Action { implicit request =>
-    val id = request.body.asFormUrlEncoded.get("token").head
+  def sentryLookup(requestType: String, memberFilter: Member => Boolean) = {
     // wtf, Adam
-    val members = (fakeMemberService.getStaff++fakeMemberService.getStudents).filter(m => m.universityId == id)
+    val members = (fakeMemberService.getStaff++fakeMemberService.getStudents).filter(memberFilter)
 
     if(members.isEmpty) {
-      Ok("returnType=51")
+      Ok("returnType=5" + requestType)
```
Comment (matmannion) [inline app/controllers/IndexController.scala:131]: omg how have I never realised this is what the error codes represent

File: app/controllers/IndexController.scala [hunk 3]
```diff
@@ -121,18 +123,34 @@ class IndexController extends BaseController {
...
     }
     else {
       val response = fakeMemberService.getResponseFor(members.head)
-      val attributes = AttributeConverter.toAttributes(response)
-      Ok("returnType=1\nid=" + members.head.universityId + "\n" + attributes.map(_.productIterator.mkString("=")).mkString("\n"))
+      val attributes = AttributeConverter.toAttributes(response, false)
```
Comment (matmannion) [inline app/controllers/IndexController.scala]: Named call here, please

File: app/controllers/IndexController.scala [hunk 4]
```diff
@@ -121,18 +123,34 @@ class IndexController extends BaseController {
...
+      Ok("returnType=" + requestType + "\nid=" + members.head.universityId + "\n" + attributes.map(_.productIterator.mkString("=")).mkString("\n"))
+    }
+  }
+
+  def respondToSentry(requestType: Int, user: Option[String]) = Action { implicit request =>
+    val formData = request.body.asFormUrlEncoded;
+    
+    requestType match {
+      case 1 if (request.method == "POST" && formData.get("token").nonEmpty) =>
```
Comment (matmannion) [inline app/controllers/IndexController.scala]: Code style thing, can we remove the brackets after the if on these? Not needed for case match filters, it's effectively a .withFilter

File: app/domain/Member.scala [hunk 5]
```diff
@@ -93,30 +93,39 @@ object WarwickItsClass {
...
+      (if (oldMode) "firstname" else "givenName") -> r.givenName,
+      (if (oldMode) "lastname" else "sn") -> r.familyName,
       "title" -> r.title,
-      "passwordexpired" -> "FALSE"
+      "passwordexpired" -> "FALSE",
+      "user" -> r.userCode
     )
+
+    if (oldMode)
+      return attribs + ("name" -> (r.givenName + " " + r.familyName), "member" -> (isStaff || isStudent).toString)
```
Comment (matmannion) [inline app/domain/Member.scala]: Explicit `return` not needed, the last statement in the function is always a return

**Review by matmannion — APPROVED** [review]

**Comment by oscar-h64** [comment]
Should all be fixed in https://github.com/UniversityofWarwick/sso-stub/pull/14/commits/b86cecde947b0ef7914b13035622a12da9bd78d0

**Pull request merged**